packages/angular_devkit/build_angular/src/builders/browser-esbuild/index.ts

```diff
@@ -0,0 +1,314 @@
+
+    return { success: false };
+  }
+
+  // Process global stylesheets
+  if (options.styles) {
+    // resolveGlobalStyles is temporarily reused from the Webpack builder code
+    const { entryPoints: stylesheetEntrypoints, noInjectNames } = resolveGlobalStyles(
+      options.styles,
+      workspaceRoot,
+      !!options.preserveSymlinks,
+    );
+    for (const [name, files] of Object.entries(stylesheetEntrypoints)) {
+      const virtualEntryData = files.map((file) => `@import '${file}';`).join('\n');
+      const sheetResult = await bundleStylesheetText(
+        virtualEntryData,
+        { virtualName: `angular:style/global;${name}`, resolvePath: workspaceRoot },
+        {
+          optimization: !!optimizationOptions.styles.minify,
+          sourcemap: !!sourcemapOptions.styles,
+          outputNames: noInjectNames.includes(name) ? { media: outputNames.media } : outputNames,
+        },
+      );
+
+      await logMessages(context, sheetResult);
+      if (!sheetResult.path) {
+        // Failed to process the stylesheet
+        assert.ok(
+          sheetResult.errors.length,
+          `Global stylesheet processing for '${name}' failed with no errors.`,
+        );
+
+        return { success: false };
+      }
+
+      // The virtual stylesheets will be named `stdin` by esbuild. This must be replaced
+      // with the actual name of the global style and the leading directory separator must
+      // also be removed to make the path relative.
+      const sheetPath = sheetResult.path.replace('stdin', name).slice(1);
+      outputFiles.push(createOutputFileFromText(sheetPath, sheetResult.contents));
+      if (sheetResult.map) {
+        outputFiles.push(createOutputFileFromText(sheetPath + '.map', sheetResult.map));
+      }
+      if (!noInjectNames.includes(name)) {
+        initialFiles.push({
+          file: sheetPath,
+          name,
+          extension: '.css',
+        });
+      }
+      outputFiles.push(...sheetResult.resourceFiles);
+    }
+  }
+
+  // Generate index HTML file
+  if (options.index) {
+    const entrypoints = generateEntryPoints({
+      scripts: options.scripts ?? [],
+      styles: options.styles ?? [],
+    });
+
+    // Create an index HTML generator that reads from the in-memory output files
+    const indexHtmlGenerator = new (class extends IndexHtmlGenerator {
+      override async readAsset(path: string): Promise<string> {
+        // Remove leading directory separator
+        path = path.slice(1);
+        const file = outputFiles.find((file) => file.path === path);
+        if (file) {
+          return file.text;
+        }
+
+        throw new Error(`Output file does not exist: ${path}`);
+      }
+    })({
+      indexPath: path.join(context.workspaceRoot, getIndexInputFile(options.index)),
+      entrypoints,
+      sri: options.subresourceIntegrity,
+      optimization: optimizationOptions,
+      crossOrigin: options.crossOrigin,
+    });
```

Comment:
Nit: I suggest splitting the class definition and construction into two different statements, as is there's a lot going on here. It would also allow you to move the class definition to the top-level of the file and remove some unnecessary indents, though that might make the constructor a little more complicated.